### PR TITLE
Don't invoke nativeRender when surface has been destroyed

### PR DESCRIPTION
**Pull request by tobrun**
Closes https://github.com/mapbox/mapbox-gl-native/issues/14592 
Closes https://github.com/mapbox/mapbox-gl-native/issues/14252
Closes https://github.com/mapbox/mapbox-gl-native/issues/14463

This PR fixes a native crash when an app is being backgrounded and we were still scheduling renders (the mapview surface has been destroyed). Concrete steps to repriduce:
  - schedule a render on the main thread 
  - the surface gets destroyed on main thread 
  - we run nativeRender on render thread while the surface was already destroyed on main thread

Additionally to fixing the crash, I'm committing a configuration that allows to use a release build of the testapp as a debug build.

### CODE REVIEW

**Comment by LukasPaczos** [comment]
This is a great find! 
It does fix the issue reproduced by the test, but I still have my doubts about the use of the `hasSurface` flag. Looking through the code again I think that we should use a lock/mutex on the `hasSurface` variable instead of using an `AtomicBoolean`.

The atomicity of the variable doesn't completely guard us when the surface is destroyed after the if-check in the `onDrawFrame` and before we call/during the call to `nativeRender`.

Do you agree? If so, do you think we could incorporate those changes here?

File: platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/maps/renderer/glsurfaceview/GLSurfaceViewMapRenderer.java [hunk 1]
```diff
@@ -39,13 +38,17 @@ public GLSurfaceViewMapRenderer(Context context,
...
       }
 
       @Override
       public void surfaceDestroyed(SurfaceHolder holder) {
         super.surfaceDestroyed(holder);
-        hasSurface.set(false);
+        synchronized (lock) {
+          hasSurface = false;
+        }
         nativeReset();
```
Comment (LukasPaczos) [inline platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/maps/renderer/glsurfaceview/GLSurfaceViewMapRenderer.java]: this should also be synchronized as that's the actual critical part that manipulates the pointer

File: platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/maps/renderer/textureview/TextureViewMapRenderer.java [hunk 2]
```diff
@@ -43,7 +43,9 @@ public TextureViewMapRenderer(@NonNull Context context,
   @Override
   protected void onSurfaceCreated(GL10 gl, EGLConfig config) {
     super.onSurfaceCreated(gl, config);
```
Comment (LukasPaczos) [inline platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/maps/renderer/textureview/TextureViewMapRenderer.java]: same as above

File: platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/maps/renderer/textureview/TextureViewMapRenderer.java [hunk 3]
```diff
@@ -59,7 +61,9 @@ protected void onSurfaceChanged(GL10 gl, int width, int height) {
    */
   @Override
   protected void onSurfaceDestroyed() {
-    hasSurface.set(false);
+    synchronized (lock) {
+      hasSurface = false;
+    }
     super.onSurfaceDestroyed();
```
Comment (LukasPaczos) [inline platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/maps/renderer/textureview/TextureViewMapRenderer.java]: same as above

**Comment by tobrun** [comment]
>GLSurfaceViewMapRenderer#onSurfaceCreated, GLSurfaceViewMapRenderer#onSurfaceDestroyed and GLSurfaceViewMapRenderer#onSurfaceChanged should also be synchronised.

Added those in https://github.com/mapbox/mapbox-gl-native/pull/14821/commits/0f877c972a1dd2a737280029262b00fae0ddaa1f though the flag that we are guarding isn't set or get in those blocks. I'm actually not seeing why we would need them. Could you add an example use-case where it would fail without synchronization of those blocks?

**Comment by tobrun** [comment]
obsolete with changes made in https://github.com/mapbox/mapbox-gl-native/pull/14841

**Pull request closed without merging**